code/datums/components/surgery_initiator.dm

```diff
@@ -0,0 +1,281 @@
+/// Allows an item to be used to initiate surgeries.
+/datum/component/surgery_initiator
+	/// If present, this surgery TYPE will be attempted when the item is used.
+	/// Useful for things like limb reattachments that don't need a scalpel.
+	var/datum/surgery/forced_surgery
+
+	/// If true, the initial step will be cancellable by just using the tool again. Should be FALSE for any tool that actually has a first surgery step.
+	var/can_cancel_before_first = FALSE
+
+	/// If true, can be used with a cautery in the off-hand to cancel a surgery.
+	var/can_cancel = TRUE
+
+/**
+ * Create a new surgery initiating component.
+ *
+ * Arguments:
+ * * forced_surgery - (optional) the surgery that will be started when the parent is used on a mob.
+ */
+/datum/component/surgery_initiator/Initialize(datum/surgery/forced_surgery)
+	. = ..()
+	if(!isitem(parent))
+		return COMPONENT_INCOMPATIBLE
+
+	src.forced_surgery = forced_surgery
+
+/datum/component/surgery_initiator/Destroy(force, silent)
+	return ..()
```

Comment:
empty override? dont you need to null out `forced_surgery` here?

Reply:
Nah, `forced_surgery` should be a raw type. This is an empty override, though, and can be removed